native/src/modules/layout/containers/HeaderContainer.js

```diff
@@ -84,15 +68,15 @@ const mapStateToProps = (state: StateType, ownProps: OwnPropsType): StatePropsTy
   const routeCityModel = route ? cities.find(city => city.code === route.city) : undefined
 
   if (!route || route.status !== 'ready' || state.cities.status !== 'ready' || !state.cityContent ||
-    !languages || languages.status !== 'ready') {
+    !languages || languages.status !== 'ready' || !stateCityCode) {
     // Route does not exist yet. In this case it is not really defined whether we are peek or not because
     // we do not yet know the city of the route.
     return {
       language: state.contentLanguage,
       routeCityModel,
       peeking: false,
       categoriesAvailable: false,
-      shareUrl: null
+      shareUrl: simpleRouteShareUrl
```

Comment:
Optional: Keep the `shareUrlFromScene` above and change this to
```suggestion
      shareUrl: shareUrlFromScene || simpleRouteShareUrl
```.
It is not really needed though as the only routes with share url still in the route params will be image and pdf routes which are not using the normal header and therefore this header container.